Another attempt at getting project name from git.

thepont opened the pull request:
I didn't realize that #14 already existed before I started working on this, so it duplicates a lot of work already done in that PR.

Although it also uses `<<LAST_PROJECT>>` as opposed to putting all unknown Terminal information in the Terminal project this should _I think_ resolve #17 as well and will better associate time.

wbingli commented:
Thanks for your contributing.  It's a nice and clean change.  

I don't want to break current user behavior (always use `Terminal` project).  Instead,  can introduce an new environment  like `$WAKATIME_ENABLE_PROJECT_DECTION` if anyone want to enable this feature.

Let me know if you wanna add this support. Otherwise, I can merge your PR and add this env support.

thepont commented:
@wbingli I have added the ability to set `ZSH_WAKATIME_PROJECT_DETECTION` to enable project detection, also added some documentation in the README file about the new flag.

wbingli commented:
Awesome job!  Merged, THANKS!
